### DISCUSSION TITLE
What is proper way to set up `chainWebpack` or `configureWebpack` in plugins?

### DISCUSSION THREAD

**Discussion by joshuacurtiss** [vuepress/core#803] [Q&A] [answered]
As the [migration notes for plugin authors](https://v2.vuepress.vuejs.org/guide/migration.html#for-plugin-authors) points out, Webpack related hooks are now removed and it is suggested you modify `app.options.bundlerConfig` directly to set a `chainWebpack` or `configureWebpack` function from a plugin. 

I understand that and it works for a single plugin, but setting this directly will cause multiple instances (or just multiple plugins using this approach) to stomp on each other, and only the last one wins.

Am I missing an approach to avoid this problem? Suggestions?  Thanks for your help.

Josh

**Comment by Mister-Hope** [comment]
~~You may modify bundler config directly, if you return something, ift will be merged, and for array options, the behavior is concat.~~

Oh, I think I understand what you mean,once a plugin is added with options, you probably can not read the original options or modify that, but you can still remove that plugin and add the same plugin with options you like.

So try avoid using some bundler plugins if you have workaround (e.g. you may convert svg to a standard Vue component, so that you can avoid using `vue-svg-loader`. If you must make some edits to let your plugin work, you probably need to mention it in docs to let users set them.

We have plans improving bundler config (probably with a new option)

**Comment by meteorlxy** [comment, marked as answer]
In latest version, you could handle it via [extendsBundlerOptions](https://v2.vuepress.vuejs.org/reference/plugin-api.html#extendsbundleroptions) hook
